### PR TITLE
update apkid_rules

### PR DESCRIPTION
**Pull request by superpoussin22**
use the last apkid rules

### Describe the Pull Request

```
update apkid rules cause rule sin Pypi package can be old 
```

### Checklist for PR

- [X] Run MobSF unit tests (http://your-mobsf-@ip:8000/tests/ or python3 manage.py test).
- [X] Tested Working on Docker
- [X] Make sure build is passing on your PR. [Build Status](https://travis-ci.com/MobSF/Mobile-Security-Framework-MobSF/pull_requests)
- [X] Coding style (indentation, etc)

### Additional Comments (if any)

```
did not solve incomplete apkid result need another PR 
```

### CODE REVIEW

File: MalwareAnalyzer/views/apkid.py [hunk 1]
```diff
@@ -1,26 +1,43 @@
...
-def apkid_analysis(app_dir, apk_file):
+def apkid_analysis(app_dir, apk_file, apk_name):
     """APKiD Analysis of DEX files"""
     apkid_res = {}
     if not settings.APKID_ENABLED:
         return apkid_res
     if not os.path.exists(apk_file):
         logger.error("APKiD - APK not found")
         return {'error': True}
+    dest_dir = app_dir + "apkid/"
```
Comment (ajinabraham) [inline MalwareAnalyzer/views/apkid.py]: us os.path.join()

File: MalwareAnalyzer/views/apkid.py [hunk 2]
```diff
@@ -1,26 +1,43 @@
...
+        result = json.load(f)
     for filep, res in result.items():
-        dex_file = filep.rsplit("!", 1)[1]
+        try:
+            dex_file = filep.rsplit("!", 1)[1]
+        except Exception:
+            #dex_file = filename_from_path(filep)
+            dex_file = apk_name
         apkid_res[dex_file] = res
+    os.remove(report)
```
Comment (ajinabraham) [inline MalwareAnalyzer/views/apkid.py]: use: https://docs.python.org/3/library/shutil.html#shutil.rmtree

File: MalwareAnalyzer/views/apkid.py (hunk 1 above)
Comment (superpoussin22) [reply MalwareAnalyzer/views/apkid.py]: directory is removed at the end of the scan to preserve space

Comment (superpoussin22) [reply MalwareAnalyzer/views/apkid.py]: done

File: MalwareAnalyzer/views/apkid.py (hunk 2 above)
Comment (superpoussin22) [reply MalwareAnalyzer/views/apkid.py]: done

File: MalwareAnalyzer/views/apkid.py [hunk 3]
```diff
@@ -1,26 +1,41 @@
...
-def apkid_analysis(app_dir, apk_file):
+def apkid_analysis(app_dir, apk_file, apk_name):
     """APKiD Analysis of DEX files"""
     apkid_res = {}
     if not settings.APKID_ENABLED:
         return apkid_res
     if not os.path.exists(apk_file):
         logger.error("APKiD - APK not found")
         return {'error': True}
+    dest_dir = os.path.join(app_dir, "apkid/")
```
Comment (ajinabraham) [inline MalwareAnalyzer/views/apkid.py]: You don't have to use `apkid/`.
Just `apkid` is enough. After all we are using path.join not to worry about the delimiter.

File: MalwareAnalyzer/views/apkid.py [hunk 4]
```diff
@@ -1,26 +1,41 @@
...
+    os.mkdir(dest_dir)
+    shutil.copy2(apk_file, dest_dir)
     from apkid import apkid
     from apkid import __version__ as apkid_ver
     logger.info("Running APKiD %s on: %s", apkid_ver,
                 filename_from_path(apk_file))
-    result = apkid.scan_apk(apk_file, 30, True)
+    result = apkid.scan(dest_dir, 30, True, dest_dir, True)
+    temporary_apk = dest_dir +  filename_from_path(apk_file)
```
Comment (ajinabraham) [inline MalwareAnalyzer/views/apkid.py]: Please use path.join here as well

File: MalwareAnalyzer/views/apkid.py [hunk 5]
```diff
@@ -1,26 +1,41 @@
...
+    report = dest_dir +  filename_from_path(apk_file) +  "_apkid.json"
```
Comment (ajinabraham) [inline MalwareAnalyzer/views/apkid.py]: And here

File: MalwareAnalyzer/views/apkid.py (hunk 4 above)
Comment (ajinabraham) [reply MalwareAnalyzer/views/apkid.py]: Assign filename_from_path(apk_file) to variable and reuse the variable than calling it again

File: MalwareAnalyzer/views/apkid.py [hunk 6]
```diff
@@ -1,26 +1,41 @@
...
+    with open(report, 'r') as f:
+        result = json.load(f)
     for filep, res in result.items():
-        dex_file = filep.rsplit("!", 1)[1]
+        try:
+            dex_file = filep.rsplit("!", 1)[1]
+        except Exception:
```
Comment (ajinabraham) [inline MalwareAnalyzer/views/apkid.py]: Why try and exception?
You can do an if else to check for. Apk or. Dex right?

File: MalwareAnalyzer/views/apkid.py (hunk 3 above)
Comment (superpoussin22) [reply MalwareAnalyzer/views/apkid.py]: done

File: MalwareAnalyzer/views/apkid.py (hunk 6 above)
Comment (superpoussin22) [reply MalwareAnalyzer/views/apkid.py]: yep I could :)

**Comment by ajinabraham** [comment]
So now we have the updated rules only in docker now.

**Pull request merged**